### PR TITLE
[ONNX] Fix export of images with no detection

### PR DESCRIPTION
**Pull request by neginraoof**
Fix export of images with no detection for Mask RCNN

### CODE REVIEW

**Comment by neginraoof** [comment]
cc @fmassa for review. Thanks!

File: torchvision/models/detection/roi_heads.py [hunk 1]
```diff
@@ -322,22 +321,26 @@ def keypointrcnn_inference(x, boxes):
...
+
+        w_half = w_half.to(dtype=torch.float64) * scale
+        h_half = h_half.to(dtype=torch.float64) * scale
+
+        boxes_exp0 = x_c - w_half
+        boxes_exp1 = y_c - h_half
+        boxes_exp2 = x_c + w_half
+        boxes_exp3 = y_c + h_half
+        boxes_exp = torch.stack((boxes_exp0, boxes_exp1, boxes_exp2, boxes_exp3), 1)
+        return boxes_exp.to(dtype=boxes.dtype)
```
Comment (fmassa) [inline torchvision/models/detection/roi_heads.py:343]: Out of curiosity, which operation doesn't support empty tensors in ONNX?

File: torchvision/models/detection/roi_heads.py [hunk 2]
```diff
@@ -322,22 +321,26 @@ def keypointrcnn_inference(x, boxes):
...
-    h_half = h_half.to(dtype=torch.float32) * scale
-
-    boxes_exp0 = x_c - w_half
-    boxes_exp1 = y_c - h_half
-    boxes_exp2 = x_c + w_half
-    boxes_exp3 = y_c + h_half
-    boxes_exp = torch.stack((boxes_exp0, boxes_exp1, boxes_exp2, boxes_exp3), 1)
-    return boxes_exp
+    if boxes.numel() == 0:
+        return torch.empty(0, dtype=boxes.dtype)
```
Comment (fmassa) [inline torchvision/models/detection/roi_heads.py:328]: This will return a box with a different number of dimensions in this case, as we normally return boxes with `Nx4` dimensions. It might bring issues in the future

File: torchvision/models/detection/keypoint_rcnn.py [hunk 3]
```diff
@@ -237,7 +237,7 @@ def __init__(self, in_channels, num_keypoints):
         super(KeypointRCNNPredictor, self).__init__()
         input_features = in_channels
         deconv_kernel = 4
-        self.kps_score_lowres = misc_nn_ops.ConvTranspose2d(
+        self.kps_score_lowres = nn.ConvTranspose2d(
```
Comment (fmassa) [inline torchvision/models/detection/keypoint_rcnn.py:240]: Great that we already support empty tensors in PyTorch! Thanks for the cleanup!

**Comment by fmassa** [comment]
Apart from this question, I think the PR looks good to merge (minus the tensor having a different number of dimensions)

File: torchvision/models/detection/roi_heads.py (hunk 2 above)
Comment (neginraoof) [reply torchvision/models/detection/roi_heads.py]: Actually looks like this branch is not needed. I have cleaned this up.

File: torchvision/models/detection/roi_heads.py (hunk 1 above)
Comment (neginraoof) [reply torchvision/models/detection/roi_heads.py]: So I was seeing an issue when broadcasting And op:
"Can broadcast 0 by 0 or 1. 3 is invalid"
But after looks like this is not needed anymore. I've reverted this change.

**Comment by neginraoof** [comment]
@fmassa Thanks a lot for pointing out the changes in roi_heads.
If you agree, I can go ahead and remove class ConvTranspose2d() for misc.py since it's not used in the code anymore.

File: torchvision/ops/misc.py [hunk 4]
```diff
@@ -16,59 +16,6 @@
...
-            raise ValueError('Only `zeros` padding mode is supported for ConvTranspose2d')
-
-        output_padding = self._output_padding(input, output_size, self.stride, self.padding, self.kernel_size)
-
-        return F.conv_transpose2d(
-            input, self.weight, self.bias, self.stride, self.padding,
-            output_padding, self.groups, self.dilation)
-
-
-class BatchNorm2d(torch.nn.BatchNorm2d):
```
Comment (fmassa) [inline torchvision/ops/misc.py:60]: This is technically a BC-breaking change, as if users were relying on `ConvTranspose2d` or `BatchNorm2d` from `misc.py` this will error now. I'll merge this as is to move forward, but I'll send a follow-up PR adding back a backward-compatibility fix.

Comment (neginraoof) [reply torchvision/ops/misc.py:60]: Got it. Thanks a lot @fmassa
